### ISSUE TITLE
Custom Indicator is not centred like default indicator type

### ISSUE THREAD

**Issue by AhmedKhattak** [onevcat/Kingfisher#797]
### Check List

Thanks for considering to open an issue. Before you submit your issue, please confirm these boxes are checked.

- [✓ ] I have read the [wiki page](https://github.com/onevcat/Kingfisher/wiki) and [cheat sheet](https://github.com/onevcat/Kingfisher/wiki/Cheat-Sheet), but there is no information I need.
- [✓] I have searched in [existing issues](https://github.com/onevcat/Kingfisher/issues?utf8=✓&q=is%3Aissue), but did not find a same one.
- [✓] I want to reoprt a problem instead of asking a question. It'd better to use [kingfisher tag in Stack Overflow](http://stackoverflow.com/questions/tagged/kingfisher) to ask a question.

### Issue Description
Custom Indicator is not centred like default indicator type

#### What
The custom indicator UIView shows up on top left side of the corner always
#### Reproduce

1. Make a custom indicator using UIView subclass
2. Conform UIView Class to Indicator Protocol
3. Initialise UIView using CGRect(x: 99 , y: 99, width: 30, height: 30)
4. And set the image view indicator to .custom(indicator: bounceLoader)

##### BounceLoader.swift
```
import UIKit
import Kingfisher

class BounceLoader: UIView {
    
    var outerCirlce: UIView = UIView()
    var innerCircle: UIView = UIView()
    
    
    let transformOne = CGAffineTransform(scaleX: 0.01, y: 0.01)
    let transformTwo = CGAffineTransform(scaleX: 1.6, y: 1.6)
    
    
    override init(frame: CGRect) {
        super.init(frame: frame)
        commonInit()
    }
    
    required init?(coder aDecoder: NSCoder) {
        super.init(coder: aDecoder)
        commonInit()
    }
    
    
    func commonInit() {
        
        self.backgroundColor = UIColor.clear
        
        outerCirlce.frame = bounds
        innerCircle.frame = bounds
        
        
        
        
        
        makeUIViewCircle(view: outerCirlce)
        makeUIViewCircle(view: innerCircle)
        
        makeUIViewTranslucent(view: outerCirlce, blurStyle: .dark)
        makeUIViewTranslucent(view: innerCircle, blurStyle: .light)
        
        addSubview(outerCirlce)
        addSubview(innerCircle)
        
        startAnimation()
    }
    
    
    
    private func makeUIViewCircle(view: UIView) {
        view.layer.cornerRadius = view.frame.width/2
    }
    
    
    private func makeUIViewTranslucent(view: UIView, blurStyle: UIBlurEffectStyle) {
        view.backgroundColor = UIColor.clear
        let blurEffect = UIBlurEffect(style: blurStyle)
        let uiVisualEffect = UIVisualEffectView(effect: blurEffect)
        uiVisualEffect.frame = view.bounds
        uiVisualEffect.autoresizingMask = [.flexibleWidth, .flexibleHeight]
        view.clipsToBounds = true
        view.addSubview(uiVisualEffect)
    }
    
    
    func startAnimation() {
        self.isHidden = false
        UIView.animate(withDuration: 0.6, delay: 0.0, options: [.autoreverse,.repeat,.curveEaseOut,.beginFromCurrentState,.preferredFramesPerSecond60], animations: { [weak self] in
            
            guard let `self` = self else {
                return
            }
            
            self.innerCircle.transform = self.transformOne
            self.outerCirlce.transform = self.transformTwo
            
            
            }, completion: nil)
    }
    
    
    func stopAnimation(){
        self.isHidden = true
        
    }
    
    
    
}

```

##### MyIndicator.swift
```
struct MyIndicator: Indicator {
    
    let loader = BounceLoader(frame: CGRect(x: 0.0, y: 0.0, width: 30.0, height: 30.0))
    var view: IndicatorView {
        return loader
    }
    
    func startAnimatingView() { loader.startAnimation()() }
    func stopAnimatingView() { loader.stopAnimation() }
    
}

```

#### ViewController.swift

```
let cell = collectionView.dequeueReusableCell(withReuseIdentifier: photoViewCellIdentifier, for: indexPath) as! PhotoCollectionViewCell
            
let url = URL(string: photo.urls.regular)
cell.photoImageView.backgroundColor = UIColor.hexStringToUIColor(hex: photo.color)
            
cell.photoImageView.kf.indicatorType = .custom(indicator: MyIndicator())
            
cell.photoImageView.kf.setImage(with: url , options: [.transition(.fade(0.4))])
            
return cell

```

**Comment by onevcat** [comment]
I think I found an issue in setting indicator frame. I pushed a fix on it. could you try whether it could work for you? See here: https://github.com/onevcat/Kingfisher/pull/798

**Comment by KimiChiu** [comment]
I have another issue.
I just updated to the latest version to support Swift 4.
But the indicator is not working anymore after then.
I tried to set breakpoint to the `draw` method, but it didn't get called at all.
Setting the backgroundColor to the indicatorView is not working neither.

```
var indicator = ProgressIndicator()
previewImage.isHidden = false
previewImage.clipsToBounds = true
previewImage.kf.indicatorType = .custom(indicator: indicator)
previewImage.kf.setImage(with: url, options: [.transition(.fade(0.2)), .backgroundDecode], progressBlock:{
                receivedSize, totalSize in
                let level = (Float(receivedSize) / Float(totalSize)) * 10000.0
                
                indicator.level = CGFloat(level)
            }){
                (image, error, cacheType, imageUrl) in
                if error != nil {
                    self.previewImage.kf.setImage(with: nil, placeholder: ErrorImagePlaceholder())
                }
            }
```

```
import Foundation
import UIKit
import Kingfisher

struct ProgressIndicator: Indicator {
    var progressView: ProgressView = ProgressView()
    
    var view: IndicatorView {
        return progressView
    }
    
    var viewCenter: CGPoint {
        get {
            return view.center
        }
        set {
            view.center = newValue
        }
    }
    
    var level: CGFloat{
        get{
            return progressView.level
        }
        set{
            progressView.level = newValue
        }
    }
    
    func startAnimatingView() {
        view.isHidden = false
    }
    func stopAnimatingView() {
        view.isHidden = true
    }
    
    init() {
        view.backgroundColor = UIColor(red: 0, green: 0, blue: 0, alpha: 1)
        view.isOpaque = true
        view.autoresizingMask = [.flexibleWidth, .flexibleHeight]
    }
    
}
```

```
import UIKit

class ProgressView: UIView {
    
    var mBackgroundColor = UIColor(red: 27/255, green: 94/255, blue: 32/255, alpha: 1)
    var mForegroundColor = UIColor(red: 100/255, green: 221/255, blue: 23/255, alpha: 1)
    var mRadius:CGFloat = 30.0
    var mBarWidth:CGFloat = 5
    
    public var level:CGFloat = 0.0 {
        didSet{
            setNeedsDisplay()
        }
    }
    
    var startAngle = CGFloat(Double.pi * 1.5)
    var unitAngle = CGFloat(2 * Double.pi / 10000)
    
    var mCenter: CGPoint{
        get{
            return CGPoint(x: bounds.midX, y: bounds.midY)
        }
    }
    
    func drawCircle(level: CGFloat, color: UIColor){
        let path = UIBezierPath(arcCenter: mCenter, radius: mRadius, startAngle: startAngle, endAngle: startAngle + unitAngle * level, clockwise: true)
        
        path.lineWidth = mBarWidth
        color.setStroke()
        path.stroke()
    }
    
    // Only override draw() if you perform custom drawing.
    // An empty implementation adversely affects performance during animation.
    override func draw(_ rect: CGRect) {
        drawCircle(level: 10000, color: mBackgroundColor)
        drawCircle(level: level, color: mForegroundColor)
    }

}
```

**Comment by Reelevant** [comment]
This issue is not solved in current version 4.7.0
